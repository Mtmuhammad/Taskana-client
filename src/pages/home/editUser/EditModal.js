// Modal to edit a user

import useAxiosPrivate from "../../../hooks/useAxiosPrivate";
import "../../projects/projectCard/ProjectCard.scss";
import { React, useState } from "react";

const EditModal = ({ setUsers, user, setSuccess, setErrMsg }) => {
  const axiosPrivate = useAxiosPrivate();
  const [values, setValues] = useState({});

  // set input values in state
  const onChange = (e) => {
    setValues({ ...values, [e.target.name]: e.target.value });
  };

  // handle form submission to edit user
  const handleEdit = async (e) => {
    let isMounted = true;
    const controller = new AbortController();
    e.preventDefault();
    let formData = { ...values };
    if (formData.isAdmin) formData.isAdmin = formData.isAdmin === true;

    try {
      await axiosPrivate.patch(
        `/users/${user?.empNumber}`,

        JSON.stringify(formData),
        {
          headers: { "Content-Type": "application/json" },
          withCredentials: true,
        }
      );
      const result = await axiosPrivate.get(`/users`, {
        signal: controller.signal,
      });

      setErrMsg("");
      setSuccess("User Modified!");
      isMounted && setUsers(result.data.users);
    } catch (err) {
      setErrMsg(err?.response?.data?.error?.message);
    }

    return () => {
      isMounted = false;
      controller.abort();
    };
  };

  return (
    <div
      style={{ color: "black" }}
      className="modal fade"
      id={`editUser${user?.empNumber}`}
      tabIndex="-1"
      aria-hidden="true"
      data-testid="edit-modal"
    >
      <div className="modal-dialog modal-dialog-centered modal-md modal-dialog-scrollable">
        <div className="modal-content">
          <div className="modal-header">
            <h5
              data-testid="header"
              className="modal-title  fw-bold"
              id="editProjectLabel"
            >
              {" "}
              Edit User
            </h5>
            <button
              type="button"
              className="btn-close"
              data-bs-dismiss="modal"
              aria-label="Close"
              data-testid="close-btn"
            ></button>
          </div>
          <div className="modal-body">
            <div className="mb-3">
              <label htmlFor="firstName" className="form-label">
                First Name
              </label>
              <input
                type="text"
                className="form-control"
                id={`firstName${user?.empNumber}`}
                name="firstName"
                placeholder="What is the user's first name?"
                onChange={onChange}
                data-testid="input"
                defaultValue={user?.firstName}
              />
            </div>
            <div className="mb-3">
              <label htmlFor="lastName" className="form-label">
                Last Name
              </label>
              <input
                type="text"
                className="form-control"
                id={`lastName${user?.empNumber}`}
                name="lastName"
                placeholder="What is the user's last name?"
                onChange={onChange}
                data-testid="input"
                defaultValue={user?.lastName}
              />
            </div>

            <div className="mb-3">
              <label htmlFor="email" className="form-label">
                E-mail
              </label>
              <input
                autoComplete="off"
                type="email"
                className="form-control"
                id={`email${user?.empNumber}`}
                placeholder="Add user email"
                name="email"
                onChange={onChange}
                data-testid="input"
                defaultValue={user?.email}
              ></input>
            </div>

            <div className="mb-3">
              <label htmlFor="empRole" className="form-label">
                Employee Role
              </label>
              <select
                onChange={onChange}
                className="form-select"
                id="empRole"
                name="empRole"
                defaultValue={user?.empRole}
              >
                <option value="">Choose a role:</option>
                <option value="UI/UX Designer">UI/UX Designer</option>
                <option value="Quality Assurance">Quality Assurance</option>
                <option value="Backend Developer">Backend Developer</option>
                <option value="Web Designer">Web Designer</option>
                <option value="Mobile Developer">Mobile Developer</option>
                <option value="Software Engineer">Software Engineer</option>
              </select>
            </div>
            <div className="mb-3">
              <label htmlFor="isAdmin" className="form-label">
                Is this employee admin?
              </label>
              <select
                onChange={onChange}
                className="form-select"
                id="isAdmin"
                name="isAdmin"
                defaultValue={user?.isAdmin}
              >
                <option value="" defaultValue>
                  Choose a value:
                </option>
                <option value="true">Yes</option>
                <option value="false">No</option>
              </select>
            </div>
          </div>
          <div className="modal-footer">
            <button
              type="button"
              className="fs-5 btn btn-secondary"
              data-bs-dismiss="modal"
              data-testid="cancel-btn"
            >
              Cancel
            </button>
            <button
              onClick={handleEdit}
              type="button"
              className="fs-5 btn btn-success"
              data-bs-dismiss="modal"
              data-testid="edit-btn"
            >
              Edit User
            </button>
          </div>
        </div>
      </div>
    </div>
  );
};

export default EditModal;
