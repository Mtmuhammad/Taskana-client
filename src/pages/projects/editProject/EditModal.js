// Modal to edit a new project

import useAxiosPrivate from "../../../hooks/useAxiosPrivate";
import "../projectCard/ProjectCard.scss";
import { React, useState, useEffect } from "react";
import { useNavigate, useLocation } from "react-router-dom";

const EditModal = ({
  info,
  setInfo,
  setProjects,
  project,
  setSuccess,
  setErrMsg,
}) => {
  const navigate = useNavigate();
  const axiosPrivate = useAxiosPrivate();
  const location = useLocation();

  const [values, setValues] = useState({});
  const [currProject, setCurrProject] = useState(info);

  // get single project info
  useEffect(() => {
    let isMounted = true;
    const controller = new AbortController();

    const getProject = async () => {
      try {
        const res = await axiosPrivate.get(`/projects/${project?.id}`, {
          signal: controller.signal,
        });

        isMounted && setCurrProject(res.data.project);
      } catch (err) {
        console.log(err);
        navigate("/login", { state: { from: location }, replace: true });
      }
    };

    getProject();

    return () => {
      isMounted = false;
      controller.abort();
    };
  }, [project?.id, axiosPrivate, location, navigate]);

  // set input values in state
  const onChange = (e) => {
    setValues({ ...values, [e.target.name]: e.target.value });
  };

  // handle form submission to edit project
  const handleEdit = async (e) => {
    let isMounted = true;
    const controller = new AbortController();
    e.preventDefault();
    let formData = { ...values };

    try {
      await axiosPrivate.patch(
        `/projects/${info?.id}`,

        JSON.stringify(formData),
        {
          headers: { "Content-Type": "application/json" },
          withCredentials: true,
        }
      );
      const result = await axiosPrivate.get(`/projects`, {
        signal: controller.signal,
      });

      const projectRes = await axiosPrivate.get(`/projects/${info?.id}`, {
        signal: controller.signal,
      });

      setErrMsg("");
      setSuccess("Project Modified!");
      setProjects();
      isMounted && setProjects(result.data.projects);
      setInfo();
      setInfo(projectRes.data.project);
      setValues({});
    } catch (err) {
      setErrMsg(err?.response?.data?.error?.message);
    }

    return () => {
      isMounted = false;
      controller.abort();
    };
  };

  // format date for input default value
  const formatDate = (project) => {
    const [month, day, year] = project?.deadline?.split("-");

    return [year, month, day].join("-");
  };

  return (
    <div
      className="modal fade"
      id={`editProject${project?.id}`}
      tabIndex="-1"
      aria-hidden="true"
      data-testid="edit-modal"
    >
      <div className="modal-dialog modal-dialog-centered modal-md modal-dialog-scrollable">
        <div className="modal-content">
          <div className="modal-header">
            <h5
              data-testid="header"
              className="modal-title  fw-bold"
              id="editProjectLabel"
            >
              {" "}
              Edit Project
            </h5>
            <button
              type="button"
              className="btn-close"
              data-bs-dismiss="modal"
              aria-label="Close"
              data-testid="close-btn"
            ></button>
          </div>
          <div className="modal-body">
            <div className="mb-3">
              <label htmlFor="name" className="form-label">
                Project Name
              </label>
              <input
                type="text"
                className="form-control"
                id="name"
                name="name"
                defaultValue={project?.name}
                onChange={onChange}
                data-testid="input"
              />
            </div>
            <div className="deadline-form">
              <form>
                <div className="row g-3 mb-3">
                  <div className="col">
                    <label htmlFor="deadline" className="form-label">
                      Deadline
                    </label>
                    <input
                      type="date"
                      className="form-control"
                      id="deadline"
                      defaultValue={formatDate(project)}
                      name="deadline"
                      onChange={onChange}
                      data-testid="input"
                    />
                  </div>
                </div>
              </form>
            </div>
            <div className="row g-3 mb-3">
              <div className="col-sm">
                <label htmlFor="status" className="form-label">
                  Status
                </label>
                <select
                  className="form-select"
                  aria-label="Default select Priority"
                  onChange={onChange}
                  name="status"
                  data-testid="input"
                >
                  <option defaultValue={project?.status}>
                    {project?.status}
                  </option>
                  <option value="Open">Open</option>
                  <option value="In Progress">In Progress</option>
                  <option value="Closed">Closed</option>
                </select>
              </div>
            </div>
            <div className="mb-3">
              <label htmlFor="description" className="form-label">
                Description (optional)
              </label>
              <textarea
                className="form-control"
                id="description"
                rows="5"
                defaultValue={info && info?.description}
                onChange={onChange}
                name="description"
                data-testid="input"
              ></textarea>
            </div>
          </div>
          <div className="modal-footer">
            <button
              type="button"
              className="fs-5 btn btn-secondary"
              data-bs-dismiss="modal"
              data-testid="cancel-btn"
            >
              Cancel
            </button>
            <button
              onClick={handleEdit}
              type="button"
              className="fs-5 btn btn-success"
              data-bs-dismiss="modal"
              data-testid="edit-btn"
            >
              Change Project
            </button>
          </div>
        </div>
      </div>
    </div>
  );
};

export default EditModal;
