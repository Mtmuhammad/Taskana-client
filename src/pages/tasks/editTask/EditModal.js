// Modal to edit a new task

import useAxiosPrivate from "../../../hooks/useAxiosPrivate";
import { React, useState } from "react";
import "../taskItem/TaskItem.scss";
import useAuth from "../../../hooks/useAuth";

const EditModal = ({
  task,
  setShowTasks,
  setUserTasks,
  setSuccess,
  setErrMsg,
}) => {
  const { auth } = useAuth();
  const axiosPrivate = useAxiosPrivate();
  const [values, setValues] = useState({});

  // set input values in state
  const onChange = (e) => {
    setValues({ ...values, [e.target.name]: e.target.value });
  };

  // handle form submission to edit task
  const handleEdit = async (e) => {
    let isMounted = true;
    const controller = new AbortController();
    e.preventDefault();
    let formData = { ...values };
    if (formData.important) formData.important = formData.important === "true";
    try {
      await axiosPrivate.patch(
        `/tasks/${task?.id}`,

        JSON.stringify(formData),
        {
          headers: { "Content-Type": "application/json" },
          withCredentials: true,
        }
      );
      const res = await axiosPrivate.get(`/tasks/${auth?.user?.empNumber}`, {
        signal: controller.signal,
      });

      setErrMsg("");
      setSuccess("Task Modified!");
      isMounted && setUserTasks(res.data.tasks);
      setShowTasks(res.data.tasks);
    } catch (err) {
      setErrMsg(err?.response?.data?.error?.message);
    }

    return () => {
      isMounted = false;
      controller.abort();
    };
  };

  return (
    <div
      style={{ color: "black" }}
      className="modal fade"
      id={`editTask${task?.id}`}
      tabIndex="-1"
      aria-hidden="true"
      data-testid="edit-modal"
    >
      <div className="modal-dialog modal-dialog-centered modal-md modal-dialog-scrollable">
        <div className="modal-content">
          <div className="modal-header">
            <h5
              data-testid="header"
              className="modal-title  fw-bold"
              id="editTaskLabel"
            >
              {" "}
              Edit Task
            </h5>
            <button
              type="button"
              className="btn-close"
              data-bs-dismiss="modal"
              aria-label="Close"
              data-testid="close-btn"
            ></button>
          </div>
          <div className="modal-body">
            <div className="mb-3">
              <label htmlFor="name" className="form-label">
                Task Title
              </label>
              <input
                type="text"
                className="form-control"
                id="title"
                name="title"
                placeholder="What is the title of this task?"
                onChange={onChange}
                data-testid="input"
                defaultValue={task?.title}
              />
            </div>

            <div className="mb-3">
              <label htmlFor="createTaskDescription" className="form-label">
                Description (optional)
              </label>
              <textarea
                className="form-control"
                id="createTaskDescription"
                rows="3"
                placeholder="Add any extra details about this task"
                name="description"
                onChange={onChange}
                data-testid="input"
                defaultValue={task?.description}
              ></textarea>
            </div>

            <div className="mb-3">
              <label htmlFor="createTaskImportant">Is this task urgent?</label>
              <select
                className="form-control"
                id="createTaskImportant"
                name="important"
                style={{ fontSize: "1.3rem" }}
                onChange={onChange}
                data-testid="input"
                defaultValue={task?.important}
              >
                <option default value="">
                  Choose an option:
                </option>
                <option value="true">Yes</option>
                <option value="false">No</option>
              </select>
            </div>
          </div>
          <div className="modal-footer">
            <button
              type="button"
              className="fs-5 btn btn-secondary"
              data-bs-dismiss="modal"
              data-testid="cancel-btn"
            >
              Cancel
            </button>
            <button
              data-bs-dismiss="modal"
              type="button"
              className="fs-5 btn btn-success"
              data-testid="edit-btn"
              onClick={handleEdit}
            >
              Edit Task
            </button>
          </div>
        </div>
      </div>
    </div>
  );
};

export default EditModal;
