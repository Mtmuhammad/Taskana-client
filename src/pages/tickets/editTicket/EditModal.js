// Modal to edit a new ticket

import useAxiosPrivate from "../../../hooks/useAxiosPrivate";
import { React, useState } from "react";
import useAuth from "../../../hooks/useAuth";

const EditModal = ({
  users,
  ticket,
  projects,
  setAllTickets,
  setShowTickets,
  setAssignedTickets,
  setIncompleteTickets,
  setCompleteTickets,
  setIsLoading,
  setSuccess,
  setErrMsg,
}) => {
  const { auth } = useAuth();
  const axiosPrivate = useAxiosPrivate();
  const [values, setValues] = useState({});

  // set input values in state
  const onChange = (e) => {
    setValues({ ...values, [e.target.name]: e.target.value });
  };

  // handle form submission to edit task
  const handleEdit = async (e) => {
    let isMounted = true;
    const controller = new AbortController();
    e.preventDefault();
    let formData = { ...values };
    if (formData.projectId ) formData.projectId = +formData.projectId
    if (formData.assignedTo) formData.assignedTo = +formData.assignedTo
    try {
      await axiosPrivate.patch(
        `/tickets/${ticket?.id}`,

        JSON.stringify(formData),
        {
          headers: { "Content-Type": "application/json" },
          withCredentials: true,
        }
      );
      const res = await axiosPrivate.get(`/tickets`, {
        signal: controller.signal,
      });

      let tickets = res.data.tickets.map(async (ticket) => {
        let result = await axiosPrivate.get(`/projects/${ticket.projectId}`, {
          signal: controller.signal,
        });
        let nameResult = await axiosPrivate.get(
          `/users/${ticket.assignedTo}`,
          {
            signal: controller.signal,
          }
        );
        ticket[
          "assignedName"
        ] = `${nameResult.data.user.firstName} ${nameResult.data.user.lastName}`;
        ticket["projectName"] = result.data.project.name;
        return ticket;
      });

      tickets = await Promise.all(tickets);

      isMounted && setAllTickets(tickets);
      setShowTickets()
        setShowTickets(tickets);
        setIsLoading(false)

        setAssignedTickets(
          tickets.filter(
            (ticket) => ticket.assignedTo === auth?.user?.empNumber
          ) || 0
        );
        setIncompleteTickets(
          tickets.filter((ticket) => ticket.status === "In Progress") || 0
        );
        setCompleteTickets(
          tickets.filter((ticket) => ticket.status === "Complete") || 0
        );


      setErrMsg("");
      setSuccess("Ticket Modified!");
      setValues()
    } catch (err) {
      setErrMsg(err?.response?.data?.error?.message);
    }

    return () => {
      isMounted = false;
      controller.abort();
    };
  };

  return (
    <div
      style={{ color: "black" }}
      className="modal fade"
      id={`editTicket${ticket?.id}`}
      tabIndex="-1"
      aria-hidden="true"
      data-testid="edit-modal"
    >
      <div className="modal-dialog modal-dialog-centered modal-md modal-dialog-scrollable">
        <div className="modal-content">
          <div className="modal-header">
            <h5
              data-testid="header"
              className="modal-title  fw-bold"
              id="editTicketLabel"
            >
              {" "}
              Edit Ticket
            </h5>
            <button
              type="button"
              className="btn-close"
              data-bs-dismiss="modal"
              aria-label="Close"
              data-testid="close-btn"
            ></button>
          </div>
          <div className="modal-body">
            <div className="mb-3">
              <label htmlFor="name" className="form-label">
                Ticket Title
              </label>
              <input
                type="text"
                className="form-control"
                id="title"
                name="title"
                placeholder="What is the title of this ticket?"
                onChange={onChange}
                data-testid="input"
                defaultValue={ticket?.title}
              />
            </div>

            <div className="mb-3">
              <label htmlFor="createTaskDescription" className="form-label">
                Description (optional)
              </label>
              <textarea
                className="form-control"
                id="createTaskDescription"
                rows="3"
                placeholder="Add any extra details about this ticket"
                name="description"
                onChange={onChange}
                data-testid="input"
                defaultValue={ticket?.description}
              ></textarea>
            </div>

            <div className="mb-3">
              <label htmlFor="editTicketAssignedTo">Assigned Name</label>
              <select
                className="form-control"
                id="editTicketAssignedTo"
                name="assignedTo"
                style={{ fontSize: "1.3rem" }}
                onChange={onChange}
                data-testid="input"
                defaultValue={ticket?.assignedTo}
              >
                <option default value="">
                  Choose a team member:
                </option>
                {users &&
                  users.map((user) => (
                    <option key={user?.empNumber} value={user?.empNumber}>
                      {user?.firstName} {user?.lastName} ({user.empRole})
                    </option>
                  ))}
              </select>
            </div>
            <div className="mb-3">
              <label htmlFor="editTicketProjectId">Project Name</label>
              <select
                className="form-control"
                id="editTicketProjectId"
                name="projectId"
                style={{ fontSize: "1.3rem" }}
                onChange={onChange}
                data-testid="input"
                defaultValue={ticket?.projectId}
              >
                <option default value="">
                  Choose a project:
                </option>
                {projects &&
                  projects.map((project) => (
                    <option key={project?.id} value={project?.id}>{project?.name}</option>
                  ))}
              </select>
            </div>
            <div className="mb-3">
              <label htmlFor="editTicketStatus" className="form-label">
                Ticket Status
              </label>
              <select
                id="editTicketStatus"
                className="form-select"
                onChange={onChange}
                name="status"
                data-testid="input"
                defaultValue={ticket?.status}
              >
                <option defaultValue="">Choose a status:</option>
                <option value="Open">Open</option>
                <option value="In Progress">In Progress</option>
                <option value="Complete">Complete</option>
              </select>
            </div>
          </div>
          <div className="modal-footer">
            <button
              type="button"
              className="fs-5 btn btn-secondary"
              data-bs-dismiss="modal"
              data-testid="cancel-btn"
            >
              Cancel
            </button>
            <button
              data-bs-dismiss="modal"
              type="button"
              className="fs-5 btn btn-success"
              data-testid="edit-btn"
              onClick={handleEdit}
            >
              Edit
            </button>
          </div>
        </div>
      </div>
    </div>
  );
};

export default EditModal;
