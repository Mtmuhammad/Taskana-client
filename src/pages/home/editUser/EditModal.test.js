import React from "react";
import { render, screen } from "@testing-library/react";
import EditModal from "./EditModal";
import { BrowserRouter } from "react-router-dom";

describe("smoke and snapshot tests", () => {
  test("should render without crashing", () => { 
    render(<EditModal />, { wrapper: BrowserRouter });
  });

  test("should match snapshot", () => {
    const { asFragment } = render(<EditModal />, { wrapper: BrowserRouter });
    expect(asFragment()).toMatchSnapshot();
  });
});

describe("create modal", () => {
  test("should render modal", () => {
    render(<EditModal />, { wrapper: BrowserRouter });
    const modal = screen.getByTestId("edit-modal");
    expect(modal).toBeInTheDocument();
    expect(modal).toHaveClass("modal fade");
  });
  test("should render modal header", () => {
    render(<EditModal />, { wrapper: BrowserRouter });
    const header = screen.getByTestId("header");
    expect(header).toBeInTheDocument();
    expect(header).toHaveClass("modal-title");
    expect(header).toContainHTML("Edit User");
  });
  test("should render close btn (x)", () => {
    render(<EditModal />, { wrapper: BrowserRouter });
    const close = screen.getByTestId("close-btn");
    expect(close).toBeInTheDocument();
    expect(close).toHaveClass("btn-close");
  });
  test("should render inputs", () => {
   render(<EditModal />, { wrapper: BrowserRouter });
   const inputs = screen.getAllByTestId("input");
   expect(inputs.length).toBe(3);
 });
  test("should render cancel button", () => {
    render(<EditModal />, { wrapper: BrowserRouter });
    const cancel = screen.getByTestId("cancel-btn");
    expect(cancel).toBeInTheDocument();
    expect(cancel).toHaveClass("btn-secondary");
  });
  test("should render create button", () => {
    render(<EditModal />, { wrapper: BrowserRouter });
    const cancel = screen.getByTestId("edit-btn");
    expect(cancel).toBeInTheDocument();
    expect(cancel).toHaveClass("btn-success");
  });
});
