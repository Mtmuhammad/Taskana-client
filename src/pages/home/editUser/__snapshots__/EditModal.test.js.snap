// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`smoke and snapshot tests should match snapshot 1`] = `
<DocumentFragment>
  <div
    aria-hidden="true"
    class="modal fade"
    data-testid="edit-modal"
    id="editUserundefined"
    style="color: black;"
    tabindex="-1"
  >
    <div
      class="modal-dialog modal-dialog-centered modal-md modal-dialog-scrollable"
    >
      <div
        class="modal-content"
      >
        <div
          class="modal-header"
        >
          <h5
            class="modal-title  fw-bold"
            data-testid="header"
            id="editProjectLabel"
          >
             Edit User
          </h5>
          <button
            aria-label="Close"
            class="btn-close"
            data-bs-dismiss="modal"
            data-testid="close-btn"
            type="button"
          />
        </div>
        <div
          class="modal-body"
        >
          <div
            class="mb-3"
          >
            <label
              class="form-label"
              for="firstName"
            >
              First Name
            </label>
            <input
              class="form-control"
              data-testid="input"
              id="firstNameundefined"
              name="firstName"
              placeholder="What is the user's first name?"
              type="text"
              value=""
            />
          </div>
          <div
            class="mb-3"
          >
            <label
              class="form-label"
              for="lastName"
            >
              Last Name
            </label>
            <input
              class="form-control"
              data-testid="input"
              id="lastNameundefined"
              name="lastName"
              placeholder="What is the user's last name?"
              type="text"
              value=""
            />
          </div>
          <div
            class="mb-3"
          >
            <label
              class="form-label"
              for="email"
            >
              E-mail
            </label>
            <input
              autocomplete="off"
              class="form-control"
              data-testid="input"
              id="emailundefined"
              name="email"
              placeholder="Add user email"
              type="email"
              value=""
            />
          </div>
          <div
            class="mb-3"
          >
            <label
              class="form-label"
              for="empRole"
            >
              Employee Role
            </label>
            <select
              class="form-select"
              id="empRole"
              name="empRole"
            >
              <option
                value=""
              >
                Choose a role:
              </option>
              <option
                value="UI/UX Designer"
              >
                UI/UX Designer
              </option>
              <option
                value="Quality Assurance"
              >
                Quality Assurance
              </option>
              <option
                value="Backend Developer"
              >
                Backend Developer
              </option>
              <option
                value="Web Designer"
              >
                Web Designer
              </option>
              <option
                value="Mobile Developer"
              >
                Mobile Developer
              </option>
              <option
                value="Software Engineer"
              >
                Software Engineer
              </option>
            </select>
          </div>
          <div
            class="mb-3"
          >
            <label
              class="form-label"
              for="isAdmin"
            >
              Is this employee admin?
            </label>
            <select
              class="form-select"
              id="isAdmin"
              name="isAdmin"
            >
              <option
                value=""
              >
                Choose a value:
              </option>
              <option
                value="true"
              >
                Yes
              </option>
              <option
                value="false"
              >
                No
              </option>
            </select>
          </div>
        </div>
        <div
          class="modal-footer"
        >
          <button
            class="fs-5 btn btn-secondary"
            data-bs-dismiss="modal"
            data-testid="cancel-btn"
            type="button"
          >
            Cancel
          </button>
          <button
            class="fs-5 btn btn-success"
            data-bs-dismiss="modal"
            data-testid="edit-btn"
            type="button"
          >
            Edit User
          </button>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
